packages/loader/container-loader/README.md

```diff
@@ -178,3 +181,11 @@ Hosts can also force readonly-mode for a container via calling `Container.forceR
 
 ### `storageOnly`
 Storage-only mode is a readonly mode in which the container does not connect to the delta stream and is unable to submit or recieve ops. This is useful for viewing a specific version of a document.
+
+## Dirty events
+Container runtime (IRuntime) communicates with Container about state of the container. In response, Container will raise two events - `"dirty"` and `"saved"` events. Transitions between twese two events signify presence (or lack of) user changes that were not saved to storage. In other words, if container is dirty, closing it at that moment will result in data loss from user perspective, as not all user changes made it to storage.
+This information can be used by host to build appropriate UX that allows user to be confident in the platform. For example, host may chose to show "Your changes are not saved. Are you sure you want to close document/tab/...?" kind of dialong on closing. instead (or in addition to) host may chose to show "Saving..." and "Saved" text somewhere in UX. Coupled with lack of connectvity to ordering servive (and appropriate notificaiton to the user) that may creeate enough continuos notificaiton to user not to require blocking dialog on closing.
+Please note that when active conneciton is in place, it's just a matter of time for changes to be flushed to storage (unless there is some source of continuous local changes being generated that prevents container from ever being fully saved). But if there is no active connection (let's say user is offline, or service experiences issues), then document may stay in dirty state for very long time.
```

Comment:
```suggestion
Note that when an active connection is in place, it's just a matter of time before changes will be flushed to storage unless there is some source of continuous local changes being generated that prevents container from ever being fully saved. But if there is no active connection, because the user is offline, for example, then a document may stay in a dirty state for very long time.
```